src/settings.rs

```diff
@@ -84,6 +89,7 @@ impl Default for Settings {
             profile: Some("default".to_owned()),
             overrides: BTreeMap::new(),
             pgp_keys: None,
+            auto_self_update: Some("enable".to_owned()),
```

Comment:
For a typed valued in the core here, I'd -really- rather have validated data not strings: I think it will allow fail fast rather than undefined behaviour.

Reply:
A FIXME has been added and I will change our code to support enum in the next PR, now it is not possible to use enum as a configuration item. https://github.com/rust-lang/rustup/blob/3baa99556fd2c06faaf03527a567bc026d5d0899/src/config.rs#L409

In addition this fn now validates the value entered by the user.

Reply:
I understand. I'm expressing a different opinion to @kinnison : I think converting the core to an enum is very straight forward, and there's no need to wait for another PR to do that - this PR itself sat for a week with no movement because the reviewers were busy (very sorry about that!) - but in that time the enum stage could have happened, since we all seem to be agreed that its needed.

If there is some substantial problem doing the enum work that will delay things, then I'm ok merging-and-improving, but I've looked, and it seems very shallow: have a TryInto implementation from str to the Enum type; use it in the toml parser, use it in the config setter, and you're done.

Reply:
I agree that these tasks are straightforward, but I actually want to modify the profile-related code as well. So that part of the change is not related to this pr. It only extends the pr merge cycle(It's been fifteen days.). So I would prefer to change the configuration item in the next PR.